compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -358,32 +375,30 @@ function collectNonNullsInBlocks(
       ) {
         assumedNonNullObjects.add(maybeNonNull);
       }
-      if (
-        (instr.value.kind === 'FunctionExpression' ||
-          instr.value.kind === 'ObjectMethod') &&
-        !fn.env.config.enableTreatFunctionDepsAsConditional
-      ) {
+      if (instr.value.kind === 'FunctionExpression') {
```

Comment:
we remove ObjectMethod here because any function captured in an object is assumed to be maybe not called, makes sense